go/mysql/server.go

```diff
@@ -264,7 +272,17 @@ func (l *Listener) Accept() {
 		connCount.Add(1)
 		connAccept.Add(1)
 
-		go l.handle(conn, connectionID, acceptTime)
+		go func() {
+			if l.PreHandleFunc != nil {
+				conn, err = l.PreHandleFunc(context.Background(), conn, connectionID)
```

Comment:
I'm not sure I understand the point of adding the context to the func signature if we will always pass in `context.Background()`, but I don't mind either

Reply:
The reason for passing a context is to make the function signature future proof, in case we add context support in the future. I could create this outside the for a loop to avoid creating it every single time.